### PR TITLE
feat: add support for `last_pin_timestamp`

### PR DESCRIPTION
**Pull request by Victorsitou**
## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/state.py [hunk 1]
```diff
@@ -1063,6 +1063,12 @@ def parse_channel_pins_update(self, data) -> None:
             utils.parse_time(data["last_pin_timestamp"]) if data["last_pin_timestamp"] else None
         )
 
+        try:
+            channel.last_pin_timestamp = last_pin  # type: ignore
+        except AttributeError:
+            # The channel doesn't support it.
+            pass
```
Comment (Victorsitou) [inline disnake/state.py:1070]: Another way to do this is just an `isinstance` with the supported types, but I'm not sure which is better 🤔

Comment (onerandomusername) [reply disnake/state.py:1070]: isinstance is better imo

File: disnake/threads.py [hunk 2]
```diff
@@ -129,6 +129,11 @@ class Thread(Messageable, Hashable):
         This is only available for threads created after 2022-01-09.
 
         .. versionadded:: 2.4
+
+    last_pin_timestamp: Optional[:class:`datetime.datetime`]
+        The datetime of the last pinned message or ``None`` if no message has been pinned.
```
Comment (onerandomusername) [inline disnake/threads.py:134]:
```suggestion
        The time the most recent message was pinned, or ``None`` if no message is currently pinned.
```

File: disnake/channel.py [hunk 3]
```diff
@@ -1851,15 +1860,22 @@ class DMChannel(disnake.abc.Messageable, Hashable):
         The user presenting yourself.
     id: :class:`int`
         The direct message channel ID.
+    last_pin_timestamp: Optional[:class:`datetime.datetime`]
+        The datetime of the last pinned message or ``None`` if no message has been pinned.
```
Comment (onerandomusername) [inline disnake/channel.py:1864]:
```suggestion
        The time the most recent message was pinned, or ``None`` if no message is currently pinned.
```

File: disnake/channel.py [hunk 4]
```diff
@@ -158,6 +158,11 @@ class TextChannel(disnake.abc.Messageable, disnake.abc.GuildChannel, Hashable):
         The default auto archive duration in minutes for threads created in this channel.
 
         .. versionadded:: 2.0
+
+    last_pin_timestamp: Optional[:class:`datetime.datetime`]
+        The datetime of the last pinned message or ``None`` if no message has been pinned.
```
Comment (onerandomusername) [inline disnake/channel.py:163]:
```suggestion
        The time the most recent message was pinned, or ``None`` if no message is currently pinned.
```

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**